File: src/dlls/CSSPModels/srcweb/WebMikeScenario.cs
/*
 * Manually edited
 * 
 */
using System.Collections.Generic;

namespace CSSPModels
{
    public partial class WebMikeScenario : WebBase
    {
        #region Properties
        public MikeScenario MikeScenario { get; set; }

        public List<MikeSourceModel> MikeSourceModelList { get; set; }

        public List<MikeBoundaryConditionModel> MikeBoundaryConditionModelWebTideList { get; set; }
        public List<MikeBoundaryConditionModel> MikeBoundaryConditionModelMeshList { get; set; }
        #endregion Properties

        #region Constructors
        public WebMikeScenario()
        {
            MikeScenario = new MikeScenario();
            MikeSourceModelList = new List<MikeSourceModel>();
            MikeBoundaryConditionModelWebTideList = new List<MikeBoundaryConditionModel>();
            MikeBoundaryConditionModelMeshList = new List<MikeBoundaryConditionModel>();
        }
        #endregion Constructors
    }

    public partial class MikeSourceModel : WebBase
    {
        public MikeSource MikeSource { get; set; }
        #region Properties
        public List<MikeSourceStartEnd> MikeSourceStartEndList { get; set; }
        #endregion Properties

        #region Constructors
        public MikeSourceModel()
        {
            MikeSource = new MikeSource();
            MikeSourceStartEndList = new List<MikeSourceStartEnd>();
        }
        #endregion Constructors
    }

    public partial class MikeBoundaryConditionModel : WebBase
    {
        #region Properties
        public List<MikeBoundaryCondition> MikeBoundaryConditionList { get; set; }
        #endregion Properties

        #region Constructors
        public MikeBoundaryConditionModel()
        {
            MikeBoundaryConditionList = new List<MikeBoundaryCondition>();
        }
        #endregion Constructors

    }
}

File: src/webs/CSSPWebToolsAng/client-apps/src/app/models/webmikescenario.ts
import { MikeBoundaryCondition } from './generated/MikeBoundaryCondition.model';
import { MikeScenario } from './generated/MikeScenario.model';
import { MikeSource } from './generated/MikeSource.model';
import { MikeSourceStartEnd } from './generated/MikeSourceStartEnd.model';
import { WebBase } from './webbase';

export class WebMikeScenario extends WebBase {
    TVItemParentList: WebBase[] = [];
    MikeScenario: MikeScenario = new MikeScenario();
    MikeSourceModelList: MikeSourceModel[] = [];
    MikeBoundaryConditionModelWebTideList: MikeBoundaryConditionModel[] = [];
    MikeBoundaryConditionModelMeshList: MikeBoundaryConditionModel[] = [];
}

export class MikeSourceModel extends WebBase {
    MikeSource: MikeSource = new MikeSource();
    MikeSourceStartEndList: MikeSourceStartEnd[] = [];
}

export class MikeBoundaryConditionModel extends WebBase {
    MikeBoundaryConditionList: MikeBoundaryCondition[] = [];
}